File: app/src/main/java/com/example/skogs/wifictrl/fragment/AbstractTabFragment.java
package com.example.skogs.wifictrl.fragment;

import android.content.Context;
import android.support.v4.app.Fragment;
import android.view.View;

/**
 * Created by skogs on 22.04.2016.
 */
public class AbstractTabFragment extends Fragment {

    private String title;
    protected Context context;
    protected View view;

    public void setTitle(String title) {
        this.title = title;
    }

    public String getTitle() {
        return title;
    }
}

File: app/src/main/kotlin/com/example/skogs/wifictrl/fragment/AbstractTabFragment.kt
package com.example.skogs.wifictrl.fragment

import android.content.Context
import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

/**
 * Created by skogs on 22.04.2016.
 */
abstract class AbstractTabFragment : Fragment() {

    var title: String? = null
    protected var context: Context? = null
        get() = field
        set(value) {
            field = value
        }

    protected var viewPage: View? = null

    abstract val layoutId: Int

    /* protected fun setContext(context: Context) {
         this.context = context
     }*/

    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        viewPage = inflater!!.inflate(layoutId, container, false)
        return viewPage
    }
}
